clients/docker.go

```diff
@@ -240,13 +241,21 @@ func withTcp(port string) string {
 	return fmt.Sprintf("%s/tcp", port)
 }
 
-// copyFile copies content bytes into container at /filename
-func copyFile(cli *client.Client, ctx context.Context, filename string, content []byte, containerId string) error {
+// copyFile copies content bytes into container at given file path.
+func copyFile(cli *client.Client, ctx context.Context, filePath string, content []byte, containerId string) error {
+	if len(filePath) == 0 {
+		return errors.New("zero length file path")
+	}
+	if filePath[0] != '/' {
+		filePath = "/" + filePath
+	}
+	dir, file := path.Split(filePath)
+
 	var buf bytes.Buffer
 	tw := tar.NewWriter(&buf)
 	err := tw.WriteHeader(&tar.Header{
-		Name: filename,
-		Mode: 0400,
+		Name: file,
+		Mode: 0777,
```

Comment:
0777 sounds a bit high, do we need them to be executable? thinking if it could lead to some privilege escalation vulnerabilities